core/consensus/babe/impl/babe_lottery_impl.cpp

```diff
@@ -18,27 +18,36 @@ namespace kagome::consensus {
 
   BabeLotteryImpl::BabeLotteryImpl(
       std::shared_ptr<crypto::VRFProvider> vrf_provider,
+      std::shared_ptr<primitives::BabeConfiguration> configuration,
       std::shared_ptr<crypto::Hasher> hasher)
       : vrf_provider_{std::move(vrf_provider)},
         hasher_{std::move(hasher)},
         logger_{common::createLogger("BabeLottery")} {
     BOOST_ASSERT(vrf_provider_);
     BOOST_ASSERT(hasher_);
     BOOST_ASSERT(logger_);
+    BOOST_ASSERT(configuration);
+    epoch_length_ = configuration->epoch_length;
```

Comment:
Why pass configuration if only epoch_length is needed?

Reply:
Hm... What is other way?